packages/react/src/components/__tests__/createComponent.spec.tsx

```diff
@@ -27,14 +27,24 @@ describe('createComponent - events', () => {
 });
 
 describe('createComponent - ref', () => {
-  test('should pass ref on to web component instance', () => {
+  test('should pass ref on to web component instance (RefObject)', () => {
     const ionButtonRef: React.RefObject<any> = React.createRef();
     const IonButton = createReactComponent<JSX.IonButton, HTMLIonButtonElement>('ion-button');
 
     const { getByText } = render(<IonButton ref={ionButtonRef}>ButtonNameA</IonButton>);
     const ionButtonItem = getByText('ButtonNameA');
     expect(ionButtonRef.current).toEqual(ionButtonItem);
   });
+
+  test('should pass ref on to web component instance (RefCallback)', () => {
```

Comment:
We should add a test here to test that `onIonChange` is called at most once at a time.

Reply:
Hey @liamdebeasi - I'm unsure how to go about introducing this test. I looked throughout the project and couldn't find any examples where the web components were being rendered and their behaviors verified. If you have any pointers on how to get everything set up, I'd be more than happy to add this!

Reply:
I haven't actually tested this, but I was thinking something like the following:

```jsx
let current;
const ionInputRef: React.RefCallback<any> = value => current = value;
const changeCallback = jest.fn();
const IonInput = createReactComponent<JSX.IonInput, HTMLIonInputElement>('ion-input');

const { getBy } = render(<IonInput id="my-input" onIonChange={changeCallback} ref={ionInputRef}></IonInput>);

const ionInputItem = getBy('#my-input');
ionInputItem.value = 'abc';
expect(changeCallback).toBeCalledTimes(1);
ionInputItem.value = 'abcxyz';
expect(changeCallback).toBeCalledTimes(2);
```

Reply:
Directly assigning to `.value` doesn't seem to trigger the `onIonChange` -- From what I could tell, this setup doesn't actually render the `ion-input` WebComponent.

I tried to use `fireEvent.change(ionInput, { target: { value: '' } }` but that throws "The given element does not have a value setter".

Reply:
🤔  Weird. Let's hold off on this I posted the suggestion before I saw https://github.com/ionic-team/ionic-framework/pull/23152#issuecomment-821281532, so I'll do some more digging and see if we even need to do this.